### PR TITLE
Add error handling to avoid nil dereference panic.

### PR DESCRIPTION
**Pull request by StefanHauth**

### CODE REVIEW

**Review by TimRots — APPROVED** [review]
LGTM

**Comment by TimRots** [comment]
@StefanHauth Could you please also add ``fixes #286`` to your commit message and a sign-off ?

File: marshal.go [hunk 1]
```diff
@@ -386,19 +386,29 @@ func (x *GoSNMP) send(packetOut *SnmpPacket, wait bool) (result *SnmpPacket, err
...
+				x.logPrintf("ERROR  out-of-time-window retransmit error: %s", err)
+				return nil, err
+			}
 		}
 	}
 
 	// detect unknown engine id error and retransmit with updated engine id
 	if len(result.Variables) == 1 && result.Variables[0].Name == ".1.3.6.1.6.3.15.1.1.4.0" {
-		x.logPrint("WARNING detected unknown enginer id ERROR") //nolint:misspell
+		x.logPrint("WARNING detected unknown engine id ERROR") //nolint:misspell
```
Comment (SuperQ) [inline marshal.go:400]: Can the `//nolint:misspell` be removed now?

File: marshal.go [hunk 2]
```diff
@@ -386,19 +386,29 @@ func (x *GoSNMP) send(packetOut *SnmpPacket, wait bool) (result *SnmpPacket, err
...
 		err = x.updatePktSecurityParameters(packetOut)
 		if err != nil {
 			x.logPrintf("ERROR  updatePktSecurityParameters error: %s", err)
 			return nil, err
 		}
+
 		result, err = x.sendOneRequest(packetOut, wait)
+		if err != nil {
+			x.logPrintf("ERROR  unknown engine id retransmit error: %s", err)
```
Comment (SuperQ) [inline marshal.go:409]:
```suggestion
			x.logPrintf("ERROR unknown engine id retransmit error: %s", err)
```

File: marshal.go (hunk 1 above)
Comment (StefanHauth) [reply marshal.go]: Removed

**Review by SuperQ — APPROVED** [review]

**Pull request merged**